ubg/src/search/AllFilters.js

```diff
@@ -0,0 +1,189 @@
+    if (typeof maxT === 'string') {
+      maxT = Number.MAX_SAFE_INTEGER;
+    }
+    if (minPlayer > maxP) {
+      setMaxPlayer(minPlayer);
+      maxP = minPlayer;
+    }
+    if (minTime > maxT) {
+      setMaxTime(minTime);
+      maxT = minTime;
+    }
+    ref.orderBy(sortBy, 'desc')
+        .get()
+        .then(function(querySnapshot) {
+          querySnapshot.forEach(function(doc) {
+            if (doc.data()['minPlayer'] <= maxP &&
+            minPlayer <= doc.data()['maxPlayer'] &&
+            doc.data()['minPlaytime'] <= maxT &&
+            minTime <= doc.data()['maxPlaytime'] &&
+            doc.data()['minAge'] <= minAge) {
+              list.push(doc.data());
+              if (list.length === 12) {
+                newGames.push(list);
+                list = [];
+              }
+            }
+          });
+          if (list.length !== 0) {
+            newGames.push(list);
+          }
+          setGames(newGames);
+          setPaginationCount(newGames.length);
+        })
+        .catch(function(error) {
+          console.log('Error getting documents: ', error);
+        });
+  };
+  const handleClear = () => {
+    setMinAge(21);
+    setMinPlayer(1);
+    setMaxPlayer('8+');
+    setMinTime(5);
+    setMaxTime('240+');
+    const newGames = [];
+    let list = [];
+    ref.orderBy(sortBy, 'desc')
+        .get()
+        .then(function(querySnapshot) {
+          setInitialize(true);
+          querySnapshot.forEach(function(doc) {
+            list.push(doc.data());
+            if (list.length === 12) {
+              newGames.push(list);
+              list = [];
+            }
+          });
+          if (list.length !== 0) {
+            newGames.push(list);
+          }
+          setGames(newGames);
+          setPaginationCount(newGames.length);
+          setInitialize(true);
+        })
+        .catch(function(error) {
+          console.log('Error getting documents: ', error);
+        });
+  };
+  useEffect(() => {
+    // using a hack to make useEffect act as onLoad()
+    if (initialize === false) {
+      const newGames = [];
+      let list = [];
+      ref.orderBy(sortBy, 'desc')
+          .get()
+          .then(function(querySnapshot) {
+            setInitialize(true);
+            querySnapshot.forEach(function(doc) {
+              list.push(doc.data());
+              if (list.length === 12) {
+                newGames.push(list);
+                list = [];
+              }
+            });
+            if (list.length !== 0) {
+              newGames.push(list);
+            }
+            setGames(newGames);
+            setPaginationCount(newGames.length);
+            setInitialize(true);
+          })
+          .catch(function(error) {
+            console.log('Error getting documents: ', error);
+          });
+    }
+  }, [ref, sortBy, setGames, setPaginationCount, initialize]);
+
+  return (
+    <Box boxShadow={1} m={10}>
+      <Filter label = "Minimum Age"
```

Comment:
According to our style guide I think label should be in a new line, same for all in this file.